tracer/src/Datadog.Trace/PDBs/SourceLink/GitHubSourceLinkUrlParser.cs

```diff
@@ -0,0 +1,38 @@
+﻿// <copyright file="GitHubSourceLinkUrlParser.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Linq;
+
+namespace Datadog.Trace.Pdb.SourceLink;
+
+internal class GitHubSourceLinkUrlParser : SourceLinkUrlParser
+{
+    /// <summary>
+    /// Extract the git commit sha and repository url from a GitHub SourceLink mapping string.
+    /// For example, for the following SourceLink mapping string:
+    ///     https://raw.githubusercontent.com/DataDog/dd-trace-dotnet/dd35903c688a74b62d1c6a9e4f41371c65704db8/*
+    /// It will return:
+    ///     - commit sha: dd35903c688a74b62d1c6a9e4f41371c65704db8
+    ///     - repository URL: https://github.com/DataDog/dd-trace-dotnet
+    /// </summary>
+    internal override bool ParseSourceLinkUrl(Uri uri, out string? commitSha, out string? repositoryUrl)
```

Comment:
`[NotNullWhen(true)]`

Reply:
Done in 233cf69448d27b82b4c1721ca90bb74fab58032f